test/ccall.jl

```diff
@@ -70,8 +70,10 @@ function gen_ccall_echo(x, T, U, ret=nothing)
         func_ex = :($ret($func_ex))
     end
     @gensym func_name
-    @eval @noinline $func_name(x) = $func_ex
-    :($func_name($(esc(x))))
+    quote
+        @noinline $(esc(func_name))(x) = $func_ex
```

Comment:
This now disagrees with the comment about the intended implementation of this test. But I think we can remove this change now (since the macro should run in a different world than the result post-expand)?

Reply:
Hmm, I'm not sure. I'll take a look.